### ISSUE TITLE
Release 3.2.0

### ISSUE THREAD

**Issue by daN4cat** [opensourcepos/opensourcepos#1759]
Prepare Release 3.2.0:

- close the last fixed two bugs after somebody else manages to test
- check Travis build is fine again, there are errors not due to OSPOS at the moment and last two builds failed (@jekkos ?)
- write up README and similar files
- tag master code with 3.2.0 code
- Release

The intention would be to release at the end of the month so we close another chapter.

After this release I might take a break, or at least I don't see a 3.3.0 any time soon unless other contributes new features. Of course any bug fix and hotfix release (e.g. 3.2.1) will take place soon if required.
Also languages support need a fix, at the moment the translation tool is broken and I can't fix it as I don't own it. So many a revision of 3.2.x could be just for translations.

**Comment by D3v1Lz** [comment]
That sounds sad... Thanks for what you've done so far daN4cat & everyone else...

**Comment by SteveIreland** [comment]
Purchasing is still in the queue for 3.3.  I ain't dead yet ... it just seems like it.

**Comment by daN4cat** [comment]
Yep, as REST API and Reportico integration. I will need those anyway, so I don't see how in 2018 I will not work on those....

**Comment by jekkos** [comment]
Ok I enabled repository push in weblate now

**Comment by daN4cat** [comment]
I tried 10 min ago and it didn't work. I leave to you to push otherwise we work in two on the same :-)

**Comment by daN4cat** [comment]
BTW, I managed to push Swedish?

**Comment by jekkos** [comment]
Well yes some parts of it, I think there was more, tried to get it into vcs yesterday to no avail

**Comment by daN4cat** [comment]
yes, I can see that just some translations are pushed.... hmmm. I leave to you to work on this.

**Comment by daN4cat** [comment]
It pushed just the few translations I made few min ago copy similar ones. It probably needs a retouch or something

**Comment by daN4cat** [comment]
I'll add support for Swedish and Bulgarian from the drop down menu

**Comment by jekkos** [comment]
Build seems to be fine again

**Comment by jekkos** [comment]
I just added bulgarian by copying the english files as template, seems quite tedious otherwisre, now rerunning the import for weblate, let's see if that helps

**Comment by daN4cat** [comment]
@jekkos Swedish is still not fully in master

**Comment by jekkos** [comment]
I made a ticket for this on [weblate repo](https://github.com/WeblateOrg/weblate/issues/1769). Hope they can give some advice in how to recover these

**Comment by daN4cat** [comment]
@jekkos travis keeps failing. Sometimes it's fine sometimes not. Not sure why. There is nothing strange between one code commit and the other outside....

**Comment by odiea** [comment]
Off topic . I see there is now a newer Bintray Build.

**Comment by daN4cat** [comment]
Yes but that's not the latest latest.

**Comment by jekkos** [comment]
Ok travis migrated to some newer images, that seemed to give some trouble with our setup. Build looks fine now

**Comment by daN4cat** [comment]
Still fails somehow

**Comment by daN4cat** [comment]
Lao language is a nice addition, I will add to the language list, however there is no CodeIgniter System language translation for Lao. They will need to contribute to GitHub package to get it sorted neatly.

**Comment by daN4cat** [comment]
@jekkos Travis build keeps failing

**Comment by daN4cat** [comment]
I wouldn't that it's because of who is sourcing the submission... really wondering here....

**Comment by jekkos** [comment]
Ok then I guess this guy will need to do both? Maybe we can fork CI repo for now to have it in there.

**Comment by daN4cat** [comment]
Or we can host the CI Lao language and commit to the the translation package.
Either way Weblate got stuck again with Github sync. You need to do some manual git add to get it sorted.

**Comment by mckaygerhard** [comment]
the problem her its the very complicated way of get the software pieces, git provides the submodules a better way to get the contents of the each piece of software in use in ospos.. 

so the CI must be a subrepository inside the ospos repository as modern git repository and rules must be

**Comment by daN4cat** [comment]
@jekkos there is an issue with ar-EG being ar_GE for a newly added file related to expenses. I manually moved that file and going to commit to master.

**Comment by daN4cat** [comment]
I pushed the deadline to end of January as there is one issue @objecttothis was planning to fix (Gift card with decimal issue) and we still have some Translation sync issues.
Unfortunately is difficult to keep up with fixes on one side, syncing on Weblate and etc. etc.
Also Travis builds are not stable, sometime they work, sometime they fail. It's due to packages download failures not OSPOS code itself.
@jekkos hope you can help just to sort out these couple of issues to get this done.

**Comment by objecttothis** [comment]
I'll begin working on the decimal/comma issue on or near the 18th when I return which should give plenty of time to fix the bug and test before the end of January.

**Comment by jekkos** [comment]
@daN4cat Do you know specifically which downloads are failing? I already went through the logfiles but couldn't directly spot the issue. 
Annoying this is that weblate by default uses _ for locales, while we use a hyphen. Might be easier to align the two in the end.

**Comment by daN4cat** [comment]
@jekkos not sure this is the issue but I can see these failures:

```
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/InRelease  Could not connect to ppa.launchpad.net:80 (91.189.95.83), connection timed out
W: Failed to fetch http://ppa.launchpad.net/george-edison55/cmake-3.x/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch https://packagecloud.io/computology/apt-backport/ubuntu/dists/trusty/InRelease  Failed to connect to packagecloud.io port 443: Connection timed out
W: Failed to fetch http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/git-core/ppa/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/openjdk-r/ppa/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/pollinate/ppa/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/webupd8team/java/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by daN4cat** [comment]
But I'm not sure that is the real problem here.... it's very strange. It could be a bug with Travis.

**Comment by daN4cat** [comment]
Checking a green one: https://travis-ci.org/opensourcepos/opensourcepos/builds/323669397
And a red one: https://travis-ci.org/opensourcepos/opensourcepos/builds/323777690

Actually the failure looks like being with bintray:

```
Installing deploy dependencies
[Bintray Upload] Reading descriptor file: deployment.json
[Bintray Upload] Uploading file 'dist/opensourcepos.zip' to /opensourcepos.20171231214157.master.9bf1d6.zip
[Bintray Upload] Bintray response: 201 Created. success
[Bintray Upload] Publishing version '3.2.0' of package 'opensourcepos'...
[Bintray Upload] Bintray response: 200 OK. 
```

**Comment by daN4cat** [comment]
@jekkos it's probably the bintray push that fails.

**Comment by jekkos** [comment]
Well I think it's not uploading if the build exits with a non zero exit code? So still the reason might be somewhere else as the failed build explicitly exits with 1.

**Comment by daN4cat** [comment]
Or the docker installation.... check the two at the end, that's the difference.

**Comment by jekkos** [comment]
good point will do a diff

**Comment by daN4cat** [comment]
@jekkos any progress?

**Comment by jekkos** [comment]
Ok switched to latest version of travis build images.. let's see if this works

**Comment by jekkos** [comment]
Afaik build is stable now..

**Comment by daN4cat** [comment]
Good. Any chance to push the translations?

**Comment by jekkos** [comment]
Ok done. Only swedish is not there yet

**Comment by daN4cat** [comment]
Good, hope you can push the Swedish in the end.

**Comment by daN4cat** [comment]
@jekkos Travis build seems fine now, probably the bower broken dependency had a role too in the broken build. Let's hope it's like that.

**Comment by jekkos** [comment]
Ok noticed that the build is stable again now. Still need to try last tip from weblate guy to have the missing translations synced. All contributors should have been notified if there were missing parts..

**Comment by jekkos** [comment]
Only two issues open on 3.2. @daN4cat do you know of any other bugs that need fixing before release?

**Comment by objecttothis** [comment]
#1694 likely has a larger scope than just giftcards since it is dealing with locale specific checking of numbers.  Once the code is working in the giftcard instance I recommend replicating the code, or refactoring it into something usable by multiple controllers/views anywhere currency amounts are entered.  I'll leave it up to you guys about whether we want to do that for 3.2.0 or push it to 3.3.0.  There is a workaround now in running a locale that doesn't require a comma for the decimal separator.

**Comment by objecttothis** [comment]
Also, if it isn't automatically done on some interval then the latest Russian and Azerbaijani languages should be pulled into the master from weblate.

**Comment by daN4cat** [comment]
@jekkos Weblate is down (Bad request 400)

**Comment by daN4cat** [comment]
@jekkos please check Weblate, the server is still down!

**Comment by daN4cat** [comment]
OK, I fixed the last two bugs.
If we can get this Weblate to rebase and be aligned, I would wait a couple of more weeks for any extra translation and then release 3.2.0.
Unless a new bug pops up....

**Comment by daN4cat** [comment]
Hopefully I fixed the last two bugs.

Now let's wait for a final confirmation and let's give a couple of weeks to complete translations or report any other odd bug and we can close.

No new feature will go in now onwards.

**Comment by daN4cat** [comment]
Almost there.... all clear at the moment.

**Comment by SteveIreland** [comment]
waiting ... waiting ... waiting ... :wink:

**Comment by daN4cat** [comment]
Prepared the release notes, readme and whatnew.

Just one more week for any translation....

**Comment by daN4cat** [comment]
@jekkos I managed to break Weblate again on the rebase part.
I removed migrate language files and it looks like it's not happy.
Can you please do a manual rebase of Weblate to fix the issue?
Thanks.

**Comment by jekkos** [comment]
Seems fine to me..

**Comment by daN4cat** [comment]
Yes, it's fine now.

**Comment by daN4cat** [comment]
Are we good to go?

**Comment by daN4cat** [comment]
I kbnow @jekkos is still translating his part.... not sure how long to go.

**Comment by WebShells** [comment]
@jekkos so you are causing the delay -.- !

**Comment by daN4cat** [comment]
Extra week to see if we can get those last translations in and close.

**Comment by objecttothis** [comment]
Azeri is done. Staff are working to finish the first run through of Russian.

**Comment by objecttothis** [comment]
Once we move our store I will work with our staff to correct translation mistakes but first I'd like to get the translations fully populated.

**Comment by daN4cat** [comment]
OK, do you think you can close by end of the week?

**Comment by objecttothis** [comment]
I would say that at the end of the week just go with what we have.
